corehq/warehouse/etl.py

```diff
@@ -98,18 +97,26 @@ def field_mapping(cls):
         # ( <source field>, <staging field> )
         raise NotImplementedError
 
+    def validate(self):
+        super(HQToWarehouseETLMixin, self).validate()
+        model_fields = {field.name for field in self.model_cls._meta.fields}
+        mapping_fields = {field for _, field in self.field_mapping()}
+        missing = mapping_fields - model_fields
+        if missing:
+            raise Exception('Mapping fields refer to missing model fields', missing)
```

Comment:
nit: "Model fields not found in mapping fields"
Just a different phrasing that makes it a little more explicit.

Reply:
I don't find that wording any clearer or explicit - how about 'Mapping fields not present on model'?

Reply:
that works Simon.

Reply:
```suggestion
            raise Exception('Mapping fields not present on model', missing)
```